api/source/controllers/Collection.js

```diff
@@ -551,6 +551,23 @@ module.exports.createCollectionLabel = async function (req, res, next) {
   }
 }
 
+module.exports.createCollectionLabels = async function (req, res, next) {
+  try {
+    const { collectionId, grant } = await getCollectionInfoAndCheckPermission(req, Security.ROLES.Manage)
+
+    const labelIDs = await CollectionService.createCollectionLabels(collectionId, req.body)
+
+    const responses = await Promise.all(
+      labelIDs.map(id => CollectionService.getCollectionLabelById(collectionId, id, grant))
+    )
```

Comment:
Suggest returning the array of created label names (see other comment) and calling a new query that returns the labels with those names.